### DISCUSSION TITLE
Export checkstyle or SARIF format

### DISCUSSION THREAD

**Discussion by Maeda-Naoki** [biomejs/biome#2287] [Ideas]
I would like to display Biome's Lint results on the PR using [reviewdog](https://github.com/reviewdog/reviewdog).  
I can use [reviewdog-action-biome](https://github.com/marketplace/actions/run-biome-with-reviewdog) to link Reviewdog with GitHub Actions, but not with GitLab CI.    
To enhance CI collaboration, it would be better to be able to export Lint results in checkstyle or SARIF format.

**Comment by fabasoad** [comment]
+1 for SARIF

**Reply by Netail** [reply]
SARIF is on its way: https://github.com/biomejs/biome/pull/8631

**Comment by wrick17** [comment]
I made a tiny tool that converts biome-report.json to a checkstyle xml file

https://www.npmjs.com/package/biome-json-to-checkstyle

You can output a json file using 

`biome check --reporter=json > biome-report.json` 

and then use my cli to convert it to a checkstyle xml file like 

`biome-json-to-checkstyle biome-report.json checkstyle.xml`

**Comment by ematipico** [comment]
PRs are welcome! Some of our reporters were implemented by the community

**Reply by wrick17** [reply]
Tried my best. Was a real test for Cursor too as I have no Rust knowledge. https://github.com/biomejs/biome/pull/6356

**Reply by wrick17** [reply]
Holy hell, everything is failing 🤣

**Comment by Netail** [comment]
Checkstyle has already been supported since 2.3.0 & SARIF will be supported from 2.4.0 :)
